pkg/v1/mutate/mutate.go

```diff
@@ -175,52 +175,60 @@ func extract(img v1.Image, w io.Writer) error {
 		if err != nil {
 			return fmt.Errorf("reading layer contents: %v", err)
 		}
-		tarReader := tar.NewReader(layerReader)
-		for {
-			header, err := tarReader.Next()
-			if err == io.EOF {
-				break
-			}
-			if err != nil {
-				return fmt.Errorf("reading tar: %v", err)
-			}
+		readLayer := func() error {
```

Comment:
Do you need to introduce a new function here? I think we could just defer the `layerReader.Close()` and let that happen whenever we return from `extract`.

Reply:
I wasn't sure of the exact semantics of how `defer layerReader.Close()` would work in this loop, which is what led to my extra caution of ensuring each `layerReader` got closed before the end of the loop iteration.

So just to confirm: if I do `defer layerReader.Close()` in the loop, `layerReader` will get captured immediately, meaning every `layerReader` created in the loop will get closed upon return from `extract`?

Reply:
> layerReader will get captured immediately, meaning every layerReader created in the loop will get closed upon return from extract

Yeah that's my understanding, it should be fine, but I'm going to appeal to someone with more golang trivia knowledge cc @ImJasonH

Reply:
One more question: Is there a stack/scalability problem if there are a large number of layers?

Reply:
I think that's technically possible, but in practice images won't have more than ~128 layers due to technical limitations of container runtimes. I could see this being a problem if someone is abusing container image formats for other purposes, but unless we actually run into that, I'd prefer the readability of one less level of indentation.

Reply:
Fixed.